src/index.coffee

```diff
@@ -86,7 +87,15 @@ progenyConstructor = (mode, settings = {}) ->
 		deps = []
 		dirs.forEach (dir) ->
 			paths.forEach (path) ->
-				deps.push sysPath.join dir, path
+				path = sysPath.join dir, path
+				basePath = path.slice 0, -(sysPath.extname(path).length)
+				if glob.hasMagic path
+					deps.push.apply deps, glob.sync path
```

Comment:
All of these sync methods are going to kill performance. Need to refactor to use only async methods.